R/data.table.R

```diff
@@ -1881,17 +1881,64 @@ chmatch2 <- function(x, table, nomatch=NA_integer_) {
 #    x
 #}
 
-
-as.matrix.data.table <- function(x,...)
-{
-  dm <- dim(x)
-  cn <- names(x)
+as.matrix.data.table <- function(x, rownames, ...) {
+  rn <- NULL
```

Comment:
not sure we have a style guide on this, but I note that the corresponding CRAN cheat for `[.data.table` symbols are defined in the package environment rather than the function body:

https://github.com/Rdatatable/data.table/blob/master/R/data.table.R#L11

+ side is cutting that tiny bit of overhead for use cases that might repeatedly call this method; - side is increasing potential for unintentional collisions (so if moving outside the body, perhaps use some more obscure name)

Reply:
they are defined there because they are exported. `rn` won't be used by user.

Reply:
Yeap - `rn` is internal here, it will contain the vector of rownames to put in the matrix (after all the processing in `if (!missing(rownames)) {}`. `rnc` will contain the index of the column in `x` to be dropped.